server/rest/src/main/scala/org/infinispan/rest/logging/RestAccessLoggingHandler.java

```diff
@@ -0,0 +1,71 @@
+package org.infinispan.rest.logging;
+
+import io.netty.channel.ChannelHandlerContext;
+import org.infinispan.util.logging.LogFactory;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Logging filter that can be used to output requests in a similar fashion to HTTPD log output
+ *
+ * @author wburns
+ * @since 9.0
+ */
+@Provider
+public class RestAccessLoggingHandler implements ContainerResponseFilter, ContainerRequestFilter {
+   private final static JavaLog log = LogFactory.getLog(RestAccessLoggingHandler.class, JavaLog.class);
+
+   private final static String NANO_TIME = "NanoTime";
+
+   @Context
+   private ChannelHandlerContext context;
+
+   @Override
+   public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) throws IOException {
+      // IP
+      String remoteAddress = context.channel().remoteAddress().toString();
+      // Date
+      long startNano = Long.parseLong(requestContext.getHeaderString(NANO_TIME));
+      // Request method | path | protocol
+      String requestMethod = requestContext.getMethod();
+      String uri = requestContext.getUriInfo().getPath();
+      // Status code
+      int status = responseContext.getStatus();
+      // Body request size
+      int requestSize = requestContext.getLength();
+      // Body response Size - usually -1 so we calculate below
+      int responseSize = responseContext.getLength();
+      // Netty doesn't usually set the CONTENT_LENGTH on response - check if we can
+      if (responseSize == -1 && responseContext.hasEntity()) {
+         Object entity = responseContext.getEntity();
+         if (entity instanceof byte[]) {
+            responseSize = ((byte[]) entity).length;
+            responseContext.getHeaders().addFirst(HttpHeaders.CONTENT_LENGTH, responseSize);
+         }
+      }
+      // Response time
+      long responseTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNano);
+
+      log.fatalf("%s [%s] \"%s %s\" %s %d %d %d ms", remoteAddress, responseTime, requestMethod, uri, status, requestSize,
```

Comment:
Whoops :)

On Fri, Apr 29, 2016, 8:30 AM Tristan Tarrant <notifications@github.com>
wrote:

> In
> server/rest/src/main/scala/org/infinispan/rest/logging/RestAccessLoggingHandler.java
> <https://github.com/infinispan/infinispan/pull/4270#discussion_r61569189>:
>
> > +      // Body request size
> > +      int requestSize = requestContext.getLength();
> > +      // Body response Size - usually -1 so we calculate below
> > +      int responseSize = responseContext.getLength();
> > +      // Netty doesn't usually set the CONTENT_LENGTH on response - check if we can
> > +      if (responseSize == -1 && responseContext.hasEntity()) {
> > +         Object entity = responseContext.getEntity();
> > +         if (entity instanceof byte[]) {
> > +            responseSize = ((byte[]) entity).length;
> > +            responseContext.getHeaders().addFirst(HttpHeaders.CONTENT_LENGTH, responseSize);
> > +         }
> > +      }
> > +      // Response time
> > +      long responseTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNano);
> > +
> > +      log.fatalf("%s [%s] \"%s %s\" %s %d %d %d ms", remoteAddress, responseTime, requestMethod, uri, status, requestSize,
>
> Fatal ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/infinispan/infinispan/pull/4270/files/0af83f2497c5447de462a35ca4af7815f3c2ed3a#r61569189>
>